src/main/java/players/rhea/RHEAIndividual.java

```diff
@@ -0,0 +1,177 @@
+            AbstractGameState gs = gameStates[startIndex];
+            // Perform rollout and return number of FM calls taken
+            return rollout(gs, fm, startIndex, endIndex, playerID);
+        }
+        return 0;
+    }
+
+    /**
+     * Performs a rollout with random actions from startIndex to endIndex in the individual, from root game state gs.
+     * Evaluates the final state reached and returns the number of calls to the FM.next() function.
+     * @param gs - root game state from which to start rollout
+     * @param fm - forward model
+     * @param startIndex - index in individual from which to start rollout
+     * @param endIndex - index in individual where to end rollout
+     * @param playerID - ID of player, used in state evaluation
+     * @return - number of calls to the FM.next() function
+     */
+    public int rollout(AbstractGameState gs, AbstractForwardModel fm, int startIndex, int endIndex, int playerID) {
+        length = 0;
+        int fmCalls = 0;
+        double delta = 0;
+        for (int i = 0; i < startIndex; i++) {
+            double score;
+            if (this.heuristic != null){
+                score = heuristic.evaluateState(gameStates[i+1], playerID);
+            } else {
+                score = gameStates[i+1].getHeuristicScore(playerID);
+            }
+
+            delta += Math.pow(discountFactor, i) * score;
+        }
+        for (int i = startIndex; i < endIndex; i++){
+            // Rolls from chosen index to the end, randomly changing actions and game states
+            // Length of individual is updated depending on if it reaches a terminal game state
+            if (gs.isNotTerminal()) {
+                // Copy the game state
+                AbstractGameState gsCopy = gs.copy();
+                List<AbstractAction> currentActions = fm.computeAvailableActions(gsCopy);
+                AbstractAction action = null;
+                if (currentActions.size() > 0) {
+                    action = currentActions.get(gen.nextInt(currentActions.size()));
+                }
+
+                // Advance game state with random action
+                fm.next(gsCopy, action);
+                fmCalls ++;
+
+                // If it's my turn, store this in the individual
+                boolean iAmMoving = (gameStates[i].getCurrentPlayer() == playerID);
+                if (iAmMoving) {
+                    gameStates[i + 1] = gsCopy;
+                    actions[i] = action;
+
+                    // Individual length increased
+                    length++;
+
+                    // Add value of state, discounted
+                    double score;
+                    if (this.heuristic != null){
+                        score = heuristic.evaluateState(gameStates[i+1], playerID);
+                    } else {
+                        score = gameStates[i+1].getHeuristicScore(playerID);
+                    }
+                    delta += Math.pow(discountFactor, i) * score;
```

Comment:
Is this correct - delta appears to add up the discounted value of all scores. 
So consider the case in which we have scores of 1, 2, 2, 4, 3 on successive actions should we not discount the differences (i.e. 1, 0, 2, -1) and add these up to get delta to avoid multiple-counting?

The score is the total current value, not the incremental gain in reward.